lib/types/trie.go

```diff
@@ -8,46 +8,69 @@ type trieNode struct {
 }
 
 func (t *trieNode) insert(s string) {
-	if len(s) == 0 {
-		t.isLeaf = true
-		return
-	}
+	runes := []rune(s)
 
-	// mask creation of the trie by initializing the root here
 	if t.children == nil {
-		t.children = make(map[rune]*trieNode)
+		t.children = map[rune]*trieNode{}
 	}
 
-	rStr := []rune(s) // need to iterate by runes for intl' names
-	last := len(rStr) - 1
-	if c, ok := t.children[rStr[last]]; ok {
-		c.insert(string(rStr[:last]))
-		return
+	ptr := t
+	for i := len(runes) - 1; i >= 0; i-- {
+		r := runes[i]
```

Comment:
This allocation can be removed and `runes[i]` could be reused directly.

Reply:
Changed